compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1247,12 +1253,25 @@ convertChoices env mc tplTypeCon tbinds =
 
 convertChoice :: Env -> TemplateBinds -> ChoiceData -> ConvertM TemplateChoice
 convertChoice env tbinds (ChoiceData ty expr) = do
-    TConApp _ [_, _ :-> _ :-> choiceTy@(TConApp choiceTyCon _) :-> TUpdate choiceRetTy, consumingTy, _] <- convertType env ty
+    -- The desuaged representation of a Daml Choice is a five tuple.
+    -- Constructed by mkChoiceDecls in RdrHsSyn.hs in the ghc repo.
+    -- We match against that 5-tuple expression or type in 3 places in this file.
+    -- search for string "tupleElem5"
```

Comment:
since we might add more entries into this tuple i'd suggest a different name

```suggestion
    -- search for string "choiceTupleExpr"
```

Reply:
done, thanks